### ISSUE TITLE
Swagger document generation: invalid allOf usage

### ISSUE THREAD

**Issue by sashok2k** [RSuter/NSwag#1245]
Hi

Seems i got some bug when generating swagger document.
According to https://swagger.io/specification/#schemaObject  when swagger describes properties of inherited type it should describe properties inside allOf node, but current NSwag generates all properties outside of allOf. 
Such document works fine when we use NSwag CSharp code generation toolchain, but if we use swagger codegen toolchain to generate swift or kotlin or even csharp, it will not generate properties for inherited types. 

Example of incorrect type generation:

```json
{
  "x-generator": "NSwag v11.16.0.0 (NJsonSchema v9.10.39.0 (Newtonsoft.Json v9.0.0.0))",
  "swagger": "2.0",
  "info": {
    "title": "My Title",
    "version": "1.0.0"
  },
  "consumes": [
    "application/json"
  ],
  "produces": [
    "application/json"
  ],
  "paths": {
    "/api/Test": {
      "post": {
        "tags": [
          "Test"
        ],
        "operationId": "Test_Post",
        "parameters": [
          {
            "name": "value",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/CC"
            },
            "x-nullable": true
          }
        ],
        "responses": {
          "204": {
            "description": ""
          }
        }
      }
    }
  },
  "definitions": {
    "CC": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "Address": {
          "type": "string"
        }
      },
      "allOf": [
        {
          "$ref": "#/definitions/BB"
        }
      ]
    },
    "BB": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "LastName": {
          "type": "string"
        }
      },
      "allOf": [
        {
          "$ref": "#/definitions/AA"
        }
      ]
    },
    "AA": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "FirstName": {
          "type": "string"
        }
      }
    }
  }
}
```

Example of correct document: 

```json
{
  "x-generator": "NSwag v11.16.0.0 (NJsonSchema v9.10.39.0 (Newtonsoft.Json v9.0.0.0))",
  "swagger": "2.0",
  "info": {
    "title": "My Title",
    "version": "1.0.0"
  },
  "consumes": [
    "application/json"
  ],
  "produces": [
    "application/json"
  ],
  "paths": {
    "/api/Test": {
      "post": {
        "tags": [
          "Test"
        ],
        "operationId": "Test_Post",
        "parameters": [
          {
            "name": "value",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/CC"
            },
            "x-nullable": true
          }
        ],
        "responses": {
          "204": {
            "description": ""
          }
        }
      }
    }
  },
  "definitions": {
    "CC": {
      "allOf": [
        {
          "$ref": "#/definitions/BB"
        },
        {
          "type": "object",
          "additionalProperties": false,
          "properties": {
            "Address": {
              "type": "string"
            }
          }
        }
      ]
    },
    "BB": {
      "allOf": [
        {
          "$ref": "#/definitions/AA"
        },
        {
          "type": "object",
          "additionalProperties": false,
          "properties": {
            "LastName": {
              "type": "string"
            }
          }
        }
      ]
    },
    "AA": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "FirstName": {
          "type": "string"
        }
      }
    }
  }
}
```

**Comment by RSuter** [comment]
Ref: https://github.com/RSuter/NSwag/issues/1136

**Comment by RSuter** [comment]
Working on this in the "improved-inheritance-nswag-1136" branch, in http://njsonschema.org


https://github.com/RSuter/NJsonSchema/tree/features/improved-inheritance-nswag-1136/src

**Comment by RSuter** [comment]
The big problem with this "feature" is that I have to proxy all properties from the base schema to one of the allOf schemas so that you can use the schema properties transparently...

**Comment by sashok2k** [comment]
In case if someone need to fix this issue immediately please use this temporary solution.

 ```csharp
private string FixNSwagDocument(string invalidDoc)
        {
            var invalidObj = JObject.Parse(invalidDoc);
            var allOfTokens = invalidObj.SelectTokens("definitions..allOf");

            foreach (var allOf in allOfTokens)
            {
                var objToConvert = allOf.Parent.Parent;
                var newAllOffItem = JObject.Parse("{}");
                var childs = objToConvert.Children().ToList();
                foreach (var childToken in childs)
                {
                    dynamic dChild = childToken;
                    if (dChild.Name != "allOf")
                    {
                        newAllOffItem.Add(childToken);
                        childToken.Remove();
                    }
                }
                ((JArray)allOf).Add(newAllOffItem);

                Console.WriteLine(objToConvert.ToString());
            }
            return invalidObj.ToString();
        }
```

**Comment by LeroyK** [comment]
@RSuter Any update on this? Can I do anything to help?

**Comment by RSuter** [comment]
Update: Merged into NJS, next release will contain this fix

**Comment by RSuter** [comment]
Fixed in v11.20.0
